BuildSupport/SwiftSyntax/CMakeLists.txt

```diff
@@ -1,5 +1,7 @@
 include(FetchContent)
 
+cmake_minimum_required(VERSION 3.24)
```

Comment:
Can you move this to the top level cmake file? An alternative is to just put this top level and avoid the `GLOBAL`. I don't have strong opinions here, update-checkout grabs 3.24 so depending on it seems fine too 🤷